src/file.jl

```diff
@@ -960,3 +964,30 @@ end
     end
     return
 end
+
+function File(sources::Vector; kw...)
+    isempty(sources) && throw(ArgumentError("unable to read delimited data from empty sources array"))
+    length(sources) == 1 && return File(sources[1]; kw...)
+    kws = merge(values(kw), (ntasks=1,))
+    f = File(sources[1]; kws...)
+    rows = f.rows
+    for col in f.columns
+        col.column = ChainedVector([col.column])
+    end
+    files = Vector{File}(undef, length(sources) - 1)
+    @sync for i = 2:length(sources)
+        Threads.@spawn begin
+            files[i - 1] = File(sources[i]; kws...)
+        end
+    end
+    for i = 2:length(sources)
+        f2 = files[i - 1]
+        f.names == f2.names || throw(ArgumentError("column names don't match between delimited data inputs"))
+        f.types == f2.types || throw(ArgumentError("column types don't match between delimited data inputs"))
+        rows += f2.rows
+        for j = 1:length(f.columns)
+            append!(f.columns[j].column, f2.columns[j].column)
```

Comment:
I guess this will error if the first file has complete cases and subsequent files don't cause our initial `ChainedVector{T}` won't support subsequent columns of type `Union{T, Missing}`? Is the solution to pass in a full `typemap` for all possible types of `T => Union{T, Missing}`... cause that seems a bit verbose?

Reply:
Yeah, we should probably support the `T` + `Union{T, Missing}` case pretty quickly.

Reply:
Is this something that could be resolved with `ChainedVector` type? This also impacts the `joinpartitions` solution as well.

Reply:
NOTE: This seems to be even more of an issue with the latest release with inline strings.

```julia
ArgumentError: string too large (36) to convert to InlineStrings.String31
Stacktrace:
[1] stringtoolong(T::Type, n::Int64)
@ InlineStrings ~/.julia/packages/InlineStrings/2HAHk/src/InlineStrings.jl:186
[2] InlineString
@ ~/.julia/packages/InlineStrings/2HAHk/src/InlineStrings.jl:209 [inlined]
[3] convert
@ ./strings/basic.jl:232 [inlined]
[4] convert
@ ./missing.jl:69 [inlined]
[5] getpoolidx
@ ~/.julia/packages/PooledArrays/jKDrZ/src/PooledArrays.jl:548 [inlined]
[6] push!(pv::PooledArrays.PooledVector{Union{Missing, InlineStrings.String31}, UInt32, Vector{UInt32}}, v::InlineStrings.String63)
@ PooledArrays ~/.julia/packages/PooledArrays/jKDrZ/src/PooledArrays.jl:594
[7] push!
@ ~/.julia/packages/SentinelArrays/VbnBp/src/chainedvector.jl:451 [inlined]
[8] append!(A::SentinelArrays.ChainedVector{Union{Missing, InlineStrings.String31}, PooledArrays.PooledVector{Union{Missing, InlineStrings.String31}, UInt32, Vector{UInt32}}}, B::PooledArrays.PooledVector{Union{Missing, InlineStrings.String63}, UInt32, Vector{UInt32}})
@ SentinelArrays ~/.julia/packages/SentinelArrays/VbnBp/src/chainedvector.jl:608
[9] (::TableOperations.var"#23#26"{Vector{SentinelArrays.ChainedVector}, DataFrames.DataFrameColumns{DataFrames.DataFrame}})(i::Int64)
@ TableOperations ~/.julia/packages/TableOperations/ErT2o/src/TableOperations.jl:312
[10] foreach(f::TableOperations.var"#23#26"{Vector{SentinelArrays.ChainedVector}, DataFrames.DataFrameColumns{DataFrames.DataFrame}}, itr::UnitRange{Int64})
@ Base ./abstractarray.jl:2141
[11] joinpartitions(x::Tables.Partitioner{Base.Generator{Vector{String}, Tables.var"#9#10"{S3DB.var"#220#221"}}})
```